### PR TITLE
#164798292 Create , List and Update user profile

### PR DESCRIPTION
**Pull request by e-ian**
#### What does this PR do?
- Creates, update and list user profiles
#### Description of tasks to be completed.
- Creation of a profile which is done using Django signals each time a user is registered
- Updating a profile can only be done by the owner/creator of the profile
- Listing of all profiles: can be viewed by only authenticated users
#### How should this be manually tested?
- Fetch this branch by running the command git fetch origin ft-create-user-profile-164798292 in the terminal.
- Check into this branch by running git checkout ft-create-user-profile-164798292
- Activate your virtual environment by running source venv/bin/activate
- Delete the current database by running drop database DB_NAME; in the psql shell
- Create a new database by running create database DB_NAME; in the psql shell
- Update the .env file with the database name
- Source the .env file with the updated database name
- Migrate by running python manage.py migrate
- Start the application by running python manage.py runserver
- To test the register user API endpoint, Open postman, choose the POST method and paste this registration url http://localhost:8000/api/v1/users/register/. The user profile of the registered user is created automatically at this point.
- To get the created profile, run this `GET` method in postman `http://127.0.0.1:8000/api/v1/profiles/`
- To get a single user profile, run the `GET` method with the individuals username `http://127.0.0.1:8000/api/v1/profiles/jedisupreme` in postman
- To update profile, run the `PUT` method with your username and the suffix edit `http://127.0.0.1:8000/api/v1/profiles/jedisupreme/edit`
#### What are the relevant pivotal tracker stories?
[#164798292](https://www.pivotaltracker.com/story/show/164798292)
#### Screenshots
- Getting a list of user profiles after registeration
image
- Getting a single user profile
image
- Updating a user profile
image

[Finishes #164798292]

### CODE REVIEW

**Review by patrickf949 — APPROVED** [review]
Great implementation @e-ian

**Review by anyatibrian — APPROVED** [review]
good work, I really like why your serializers and signals were implemented

**Review by Genza999 — APPROVED** [review]
LGTM

File: authors/apps/profiles/models.py [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+class Profile(models.Model):
+    """
+    class creating a data model to hold information about users
+    that is used to create the user profile
+    """
+    user = models.OneToOneField(User, on_delete=models.CASCADE)
+    bio = models.TextField(blank=True)
+    full_name = models.CharField(max_length=255, blank=True)
+    following = models.BooleanField(default=False)
+    image = models.ImageField(blank=True)
```
Comment (inno-asiimwe) [inline authors/apps/profiles/models.py]: We should store a URL for the image.

**Review by Genza999 — APPROVED** [review]
LGTM. Great work mate!

**Comment by Genza999** [comment]
Great work mate! Kindly work on coveralls too!

**Review by patrickf949 — APPROVED** [review]

**Review by inno-asiimwe — APPROVED** [review]
LGTM

**Pull request merged**